### PR TITLE
framework: fix memory leaks in scene and javacomponents

### PR DESCRIPTION
**Pull request by liaxim**
GearVRf-DCO-1.0-Signed-off-by: Mihail Marinov <m.marinov@samsung.com>

### CODE REVIEW

**Comment by NolaDonato** [comment]
Should we give all hybrid objects a reference to the Java object? Would this help with cleanup? I know it would get rid of the table of colliders in the picker.

**Comment by liaxim** [comment]
Most don't need that. Could be an optional trait of a hybrid object. Something like that is needed to avoid the current split between Scene and JavaComponents.

File: GVRf/Framework/framework/src/main/jni/objects/components/render_target.cpp [hunk 1]
```diff
@@ -92,10 +92,6 @@ RenderTarget::RenderTarget()
 
 RenderTarget::~RenderTarget()
 {
-    if (mRenderTexture)
-    {
-        delete mRenderTexture;
```
Comment (dnvasilev) [inline GVRf/Framework/framework/src/main/jni/objects/components/render_target.cpp]: Good catch

**Comment by liaxim** [comment]
Planning to merge on Thursday. https://github.com/Samsung/GearVRf/pull/1572 makes a bindShader call that needs to be updated to look like the ugly section I added to render_data.cpp:283-289 here.

**Pull request merged**